lib/ask_web/controllers/short_link_controller.ex

```diff
@@ -9,29 +9,34 @@ defmodule AskWeb.ShortLinkController do
       ShortLink
       |> Repo.get_by(hash: hash)
 
-    conn =
-      conn
-      |> assign(:skip_auth, true)
+    if link do
```

Comment:
Instead of adding an `if`, the Functional Way™ is to delegate into a function with multiple definitions.

So you'd do something like `resolve_link(conn, link)`, and have one definition `defp resolve_link(conn, nil)` that returns the 404, and another one `defp resolve_link(conn, link)` that does the happy path of actually having the link in the DB.